engines/bastion_katello/app/assets/javascripts/bastion_katello/tasks/task.factory.js

```diff
@@ -1,264 +1,112 @@
+                };
+                var type = mapping[task.result] || mapping.default;
+                return { value: task.progress * 100, type: type };
+            }
 
-            function updateTask(taskObject) {
-                runningTask.task = taskObject;
-                if (taskObject.state === 'paused' || taskObject.state === 'stopped') {
-                    resource.unregisterSearch(searchId);
-                    if (taskObject.result === 'success') {
-                        runningTask.state = 'stopped';
-                        deferred.resolve(taskObject);
-                    }
-                    if ((taskObject.result === 'error' || taskObject.result === 'warning')) {
-                        runningTask.state = 'paused';
-                        deferred.reject(taskObject.humanized.errors);
-                    }
-                } else {
-                    runningTask.state = 'running';
-                    deferred.notify(taskObject);
+            // add additional fields to the task model that help with
+            // usage in the UI templates
+            function normalizeTask(task) {
+                task.progressbar = taskProgressbar(task);
+                if (!task.humanized.errors) {
+                    task.humanized.errors = [];
+                } else if (!_.isArray(task.humanized.errors)) {
+                    task.humanized.errors = [task.humanized.errors];
                 }
             }
 
-            function pollTask(taskObject) {
-                searchId = resource.registerSearch({ 'type': 'task', 'task_id': taskObject.id }, updateTask);
-            }
+            resource.queryPaged(
+                queryParams,
+                function(res) {
+                    _.each(res.results, function(task) {
+                        normalizeTask(task);
+                    });
+                    resource.cb(res);
 
-            if (task.$promise) {
-                runningTask.state = 'starting';
-                task.$promise.then(pollTask, function (error) {
-                    var errors = [];
-                    runningTask.state = 'stopped';
-                    if (error.data.errors) {
-                        errors = error.data.errors;
-                    } else if (error.data.error) {
-                        errors = error.data.error.message;
-                    }
-                    if (!_.isArray(errors)) {
-                        errors = [errors];
-                    }
-                    deferred.reject(errors);
-                });
-            } else {
-                updateTask(task);
-                pollTask(task);
-            }
+                    pendingTimeout = $timeout(function() {
+                        resource.poll(params, resource.cb);
+                    }, 8000);
+                },
+                function() {
+                    resource.cb({
+                        'human_readable_result': 'Failed to fetch task',
```

Comment:
This will need to be translated via `translate()`.

Reply:
Does `translate()` localize the text? Also, is it a service I can require into the factory?

Reply:
Yes and yes 😄